src/Type.cpp

```diff
@@ -71,7 +72,20 @@ Type *Type::void_type = NULL;
 static vector<Type *> AllTypes;
 static vector<Type *> derived_types;
 
+static vector<FunctionAttribute*> type_attributes;
+
 //////////////////////////////////////////////////////////////////////
+
+void
+GenerateAttributes()
+{
+		vector<string> common_type_attributes = {"deprecated", "unused", "transparent_union"};
+		vector<string>::iterator itr;
+		type_attributes.push_back(new MultiValuedFunctionAttribute("visibility", FuncAttrProb, {"default", "hidden", "protected", "internal"}));
+		for(itr = common_type_attributes.begin(); itr < common_type_attributes.end(); itr++)
+			type_attributes.push_back(new BooleanFunctionAttribute(*itr, FuncAttrProb));
```

Comment:
Are you sure about `FuncAttrProb`?

Reply:
My mistake. Need to change it to `TypeAttrProb`